File: app/src/main/java/cn/surine/schedulex/base/utils/Jsons.java
package cn.surine.schedulex.base.utils;

import com.google.gson.Gson;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonParser;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by surine on 2017/5/14.
 */

public class Jsons {


    /**
     * 将Json数据解析成相应的映射对象
     *
     * @param jsonData 数据
     * @param type     类型
     */
    public static <T> T parseJsonWithGson(String jsonData, Class<T> type) {
        Gson gson = new Gson();
        T result = gson.fromJson(jsonData, type);
        return result;
    }

    /**
     * 将Json数据解析成相应的映射列表
     *
     * @param json 数据
     * @param cls  类型
     */
    public static <T> List<T> parseJsonWithGsonToList(String json, Class<T> cls) {
        Gson gson = new Gson();
        List<T> mList = new ArrayList<T>();
        JsonArray array = new JsonParser().parse(json).getAsJsonArray();
        for (final JsonElement elem : array) {
            mList.add(gson.fromJson(elem, cls));
        }
        return mList;
    }


    /**
     * 将实体类转换为String
     * @param data
     * */
    public static <T> String entityToJson(List<T> data) {
        Gson gson = new Gson();
        return gson.toJson(data);
    }

}

File: app/src/main/java/cn/surine/schedulex/base/utils/Jsons.kt
package cn.surine.schedulex.base.utils

import android.util.JsonReader
import android.webkit.JsResult
import com.google.gson.Gson
import com.google.gson.JsonParser

/**
 * Intro：
 *
 * @author sunliwei
 * @date 10/25/20 09:47
 */
object Jsons {

    val gson = Gson()

    /**
     * 将Json数据解析成相应的映射对象
     * @param jsonData 数据
     */
    inline fun <reified T> parseJsonWithGson(jsonData: String?): T {
        return gson.fromJson(jsonData, T::class.java)
    }


    /**
     * 将Json数据解析成相应的映射列表
     *
     * @param json 数据
     */
    inline fun <reified T> parseJsonWithGsonToList(json: String?): List<T>? {
        return mutableListOf<T>().apply {
            for (elem in JsonParser.parseString(json).asJsonArray) {
                add(gson.fromJson(elem, T::class.java))
            }
        }
    }


    /**
     * 将实体类转换为String
     * @param data
     */
    fun <T> entityToJson(data: List<T>?): String = gson.toJson(data)

}